pkg/tcpip/network/ipv4/ipv4.go

```diff
@@ -373,7 +373,8 @@ func (e *endpoint) HandlePacket(r *stack.Route, pkt tcpip.PacketBuffer) {
 		}
 		var ready bool
 		var err error
-		pkt.Data, ready, err = e.fragmentation.Process(hash.IPv4FragmentHash(h), h.FragmentOffset(), last, more, pkt.Data)
+		fragmentation.R = r
```

Comment:
r should just be passed as a parameter to Process or passed to the reassembler when it's created so that it can use it to send the ICMP message as required.

Reply:
Earlier, we thought of passing 'r' as parameter to Process. But this was resulting in failure of some test cases in fragmentation_test.go
Also, it involved additional effort to fix those test cases. So, we decided to introduce a new variable 'R'.